src/acb_mat/mul_threaded.c

```diff
@@ -92,9 +90,30 @@ acb_mat_mul_threaded(acb_mat_t C, const acb_mat_t A, const acb_mat_t B, slong pr
         return;
     }
 
-    num_threads = flint_get_num_threads();
-    threads = flint_malloc(sizeof(pthread_t) * num_threads);
-    args = flint_malloc(sizeof(acb_mat_mul_arg_t) * num_threads);
+    num_workers = flint_request_threads(&handles, FLINT_MAX(ar, bc) - 1);
+    num_threads = num_workers + 1;
+
+    if (num_workers < 1)
```

Comment:
I'm unsure whether this special case is really needed. We avoid a little bit of overhead but I don't think this is an issue for the situations where we actually call threaded matrix multiplication.